### PR TITLE
PFW-1339 - Update MMU2 translations 

### PR DESCRIPTION
**Pull request by 3d-gussner**
# Update translations

- [X] update all message have ////MSG_ c= r=
- [X] update MMU2 messages for translations
- [X] remove duplicate code  and messages
- [X] use original messages on serial
- [X] update language scripts
  - [X] add `--shorter` argument to `lang-check.py` to show too many rows defined in MSG_ c= r=
  - [X] add current size of translation when it fails being too big for reserved space
- [X] increased 2nd language size in `config.h` to 0x3500 or 13568 bytes for MMU2 translations
- [X] cleanup messages
  - [X] rename Bondtech to Extruder
  - [X] rename AutoDeplete to SpoolJoin
 - [x] Translate po files
   - [X] updated `lang/po/Firmware.pot`
   - [X] updated and translated German missing messages `lang/po/Firmware_de.po`
   - [x] updated and translated Dutch missing messages `lang/po/Firmware_nl.po`
   - [x] update all po files
 
# ToDo
Other po files have to be translated by the internal and community members. These will be submitted in other PRs

to be updated
|Type    |Flash used|RAM used|Flash previous|RAM previous|Flash diff|RAM diff
|----------|---------------|--------------|--------------------|------------------|-------------|-----------
|MK3S  |251340      |5731        |250156 | 5731|+1184|0
|MK3    |250718      |5739        |249560 |5739 |+1158 |0
|MK25S|220750      |5463        |219528 |5463 |+1222 |0
|MK25  |222490      |5480        |221268 |5480 |+1222 |0

### CODE REVIEW

**Comment by 3d-gussner** [comment]
@vintagepc can you please review the Dutch translations
https://github.com/prusa3d/Prusa-Firmware/pull/3599/commits/ae8d1eb5463230dff9853aaebbbe0aa7ede4f319
https://github.com/prusa3d/Prusa-Firmware/pull/3599/commits/075a80e2641978e1261c5ed751d76acdfebb9a5a

@wavexx can you please review the language script changes and https://github.com/prusa3d/Prusa-Firmware/pull/3599/commits/075a80e2641978e1261c5ed751d76acdfebb9a5a
AD if you find the time to update the Italian translation.

@DRracer If you have the time please update Czech translation

@leptun If you have the time please update the Romanian translation

File: lang/po/Firmware_nl.po
Comment (vintagepc) [inline lang/po/Firmware_nl.po]: Isn't consistent with previous usage of "idler"?

File: lang/po/Firmware_nl.po
Comment (vintagepc) [inline lang/po/Firmware_nl.po]: inconsistent with previous two translations of "idler"...

File: lang/po/Firmware_nl.po
Comment (vintagepc) [inline lang/po/Firmware_nl.po:574]: Is the EN message deliberately truncated?

File: lang/po/Firmware_nl.po
Comment (vintagepc) [inline lang/po/Firmware_nl.po:1455]: Inconsistent with future translation of "pulley"

File: lang/po/Firmware_nl.po
Comment (vintagepc) [inline lang/po/Firmware_nl.po]: not consistent with previous translation of pulley

File: lang/po/Firmware_nl.po
Comment (vintagepc) [inline lang/po/Firmware_nl.po]: check consistency here too

File: lang/po/Firmware_nl.po
Comment (vintagepc) [inline lang/po/Firmware_nl.po:2275]: consistency

**Review by vintagepc — CHANGES_REQUESTED** [review]
Needs a review pass to ensure consistency of translations for Idler/Pulley in Dutch.

**Comment by 3d-gussner** [comment]
Hello community translators, how are you doing?
As we are going forward and introduce new features we need your help.

@Hauzman @QuantumRoboticsFTC @AttilaSVK @Prime1910 @PRPA041 @ingbrzy @shatter136 @Painkiller56 @pkg2000 @trondkla

FYI:
Please checkout https://github.com/prusa3d/Prusa-Firmware/tree/MK3_MMU2/lang/po Firmware_xx.po files

- No empty translations anymore. This means if the translation is the same as the origin, please copy paste it as translation.
- Try to keep new messages within <=4, <=8 or <=12 rows and avoid one word being on the next page. This looks very odd.
- As always the future PR's need to be approved by at least two translators (same language).

Please  let me know if you have time for that with a :+1: underneath.
 
Don't hesitate to ask questions and if you need a output of your translation, please let me know.

I translated the German and Dutch (waiting for review from 2nd maintainer @vintagepc). You can use it as an example.

I hope this PR will be merged soon and you can create your PRs to prusa3d/MK3_MMU2 branch.

For the more advanced users:
- run `PF-build.sh` for MK3s and MK3 to create the `/lang/firmware_<MK3S|MK3>.map` files
- change to `lang` folder
  - check your translation with `./lang-check.py --map firmware_<MK3S|MK3>.map po/Firmware_xx.po`
  - output all translations `./lang-check.py --map firmware_<MK3S|MK3>.map po/Firmware_xx.po --information`

UPDATE: @Hauzman @QuantumRoboticsFTC @AttilaSVK @Prime1910 @PRPA041 @ingbrzy @shatter136 @Painkiller56 @pkg2000 @trondkla

This PR has been merged and the prusa3d/MK3 should be used as base to work on the translations and also the PRs should point to prusa3d/MK3.

File: lang/po/Firmware_nl.po
Comment (3d-gussner) [reply lang/po/Firmware_nl.po:574]: Yes it is

File: lang/po/Firmware_nl.po
Comment (3d-gussner) [reply lang/po/Firmware_nl.po:2275]: What should we use for Pulley and Idler in Dutch?
Thanks for the fast review.

Comment (vintagepc) [reply lang/po/Firmware_nl.po:2275]: I don't think it matters too much so long as we are consistent that pulley is translated one way and idler another.

File: Firmware/mmu2_reporting.cpp
Comment (gudnimg) [inline Firmware/mmu2_reporting.cpp]: Minor nitpick: typo in "choices"

**Review by gudnimg — APPROVED** [review]
Looks good to me. I can't review the languages though. The SpoolJoin rename and button documentation is nice. 💪

File: lang/lang-check.py
Comment (wavexx) [inline lang/lang-check.py:233]: This looks good, but does it really help?
We have also "Short translation" which warns if the translation if less than half the size of the original.

File: Firmware/messages.cpp
Comment (wavexx) [inline Firmware/messages.cpp:57]: I actually like this change. Makes more sense than to mention an actual part name that some people might not be familiar with.

File: Firmware/messages.cpp
Comment (wavexx) [inline Firmware/messages.cpp:123]: I think that we should do something better for full-screen multi-page messages. The row count mostly makes sense for messages that cannot wrap.

In other cases maybe we should use a large number or something like -1 to indicate that it can wrap without limits.

**Comment by wavexx** [comment]
Aside from comments, changes to the scripts and code look good. I'll check italian later.

File: lang/lang-check.py
Comment (3d-gussner) [reply lang/lang-check.py:233]: Yes it helps me to find way too many rows defined for translations.
If someone would use all available space (defined bye c= r=) for the translations we would get ~18816bytes to squeeze into the flash/xflash. But this is of course not possible as the xflash layout just allows 15616 bytes to be used for 2nd language.

I hope to get the rows as small/needed as possible. Avoiding novels being written on 20x4 LCD screen.

File: Firmware/mmu2_reporting.cpp
Comment (3d-gussner) [reply Firmware/mmu2_reporting.cpp]: Thanks fixed

File: Firmware/messages.cpp
Comment (3d-gussner) [reply Firmware/messages.cpp:123]: No limits ... No
We have to limit content and translators to write essential information and novels.

File: lang/lang-check.py
Comment (wavexx) [reply lang/lang-check.py:233]: Got it

**Review by vintagepc — APPROVED** [review]

**Review by DRracer — APPROVED** [review]

**Pull request merged**
